ansible_collections/arista/avd/molecule/cv_deploy/invalid_config.yml

```diff
@@ -0,0 +1,129 @@
+---
+- name: "{{ test_id | upper }} Converge - cv_deploy with Workspace Abandon"
+  hosts: SITE1_FABRIC
+  connection: local
+  gather_facts: false
+  vars:
+    cv_server: www.cv-staging.corp.arista.io
+    cv_token: "{{ lookup('env', 'CVAAS_AAWG_CI') }}"
+    cv_verify_certs: true
+    cv_skip_missing_devices: true
+    eos_config_dir: "{{ playbook_dir }}/intended/configs/test_invalid_configs"
+    structured_dir: "{{ playbook_dir }}/intended/structured_configs/test_invalid_configs"
+    intended_tag_device: avd-ci-leaf1
+    intended_tags: "{{ lookup('file', structured_dir ~ '/' ~ intended_tag_device ~ '.yml')| from_yaml }}"
+    test_id: "ws-invalid-configs"
+    cv_common_pattern: "avd-cv-deploy-{{ test_id }}"
+
+  tasks:
+    - name: "{{ test_id | upper }} Banner"
+      run_once: true
+      ansible.builtin.debug:
+        msg:
+          - "######################################################################"
+          - "### STARTING MOLECULE TEST {{ '{:<38}'.format(test_id[:38]) | upper }} ####"
+          - "######################################################################"
+
+    - name: "{{ test_id | upper }} Generate random string"
+      run_once: true
+      ansible.builtin.set_fact:
+        r: "{{ lookup('password', '/dev/null chars=ascii_lowercase,digits length=4') }}"
+
+    - name: "{{ test_id | upper }} Set deploy facts"
+      run_once: true
+      ansible.builtin.set_fact:
+        cv_workspace_name: "{{ cv_common_pattern }}-{{ r }}-converge"
+        cv_workspace_description: "{{ (cv_common_pattern + '-' + r + '-converge') | upper }}"
+        cv_change_control_name: "{{ cv_common_pattern }}-{{ r }}-converge"
+        cv_change_control_description: "{{ (cv_common_pattern + '-' + r + '-converge') | upper }}"
+        cv_register_detailed_results: true
+
+    - name: "{{ test_id | upper }} Provision with workspace Pending"
+      run_once: true
+      delegate_to: localhost
+      block:
+        - name: Trigger errors by building WS with EOS CLI errors
+          ansible.builtin.import_role:
+            name: arista.avd.cv_deploy
+          vars:
+            cv_submit_workspace: false
+      rescue:
+        - name: "{{ test_id | upper }} Display CVP result"
+          run_once: true
+          ansible.builtin.debug:
+            msg: '{{ cv_deploy_results }}'
+
+        - name: "{{ test_id | upper }} Check CVP returns"
+          run_once: true
+          ansible.builtin.assert:
+            that:
+              # assert validate config build errors
+              - >
+                cv_deploy_results.workspace.build_errors | selectattr('config_validation', 'defined') | map(attribute='config_validation') | list | flatten |
```

Comment:
Let's make the build_errors easier to pass. Instead of having to fish the relevant errors out of the list which could have different errors, I would rather have the list be of config validation errors directly. Like `cv_deploy_results.workspace.build_errors.config_validation`.

Reply:
Currently converted to the following model (as both config and image validation can have both errors and warnings):
```json
                {
                    "config_validation": {
                        "errors": [
                            {
                                "configlet_name": <str>,
                                "error_code":  <str>,
                                "error_msg":  <str>,
                                "line_num": <int>
                            },
                            {
                               ...
                            }
                        ],
                        "warnings": [
                            {
                                "configlet_name":  <str>,
                                "error_code":  <str>,
                                "error_msg":  <str>,
                                "line_num": <int>
                            },
                            {
                                ...
                            }
                        ]
                    },
                    "image_validation": {
                        "errors": [
                            {
                                "sku": <str>,
                                "error_code":  <str>,
                                "error_msg":  <str>
                            },
                            {
                               ...
                            }
                       ],
                        "image_input_error": "",
                        "warnings": [
                            {
                                "sku": <str>,
                                "warning_code":  <str>,
                                "warning_msg":  <str>
                            },
                            {
                               ...
                            }
                      ]
                    },
                    "serial_number": "DCC816CEAC4BBD6319385043AD318362",
                    "stages_states": [
                        {
                            "stage": "BUILD_STAGE_IMAGE_VALIDATION",
                            "state": "SKIPPED"
                        },
                        {
                            "stage": "BUILD_STAGE_INPUT_VALIDATION",
                            "state": "SUCCESS"
                        },
                        {
                            "stage": "BUILD_STAGE_CONFIG_VALIDATION",
                            "state": "FAIL"
                        },
                        {
                            "stage": "BUILD_STAGE_CONFIGLET_BUILD",
                            "state": "SUCCESS"
                        }
                    ]
                }
```